gpflow/config/__config__.py

```diff
@@ -51,6 +55,14 @@ def default_jitter():
     return __config._jitter
 
 
+def to_default_int(x):
+    return tf.cast(x, dtype=default_int())
+
+
+def to_default_float(x):
+    return tf.cast(x, dtype=default_float())
+
+
 def set_default_int(value_type):
     try:
         tf_dtype = tf.as_dtype(value_type)  # Test that it's a tensorflow-valid dtype
```

Comment:
I know it's not part of the changes in this PR but wanted to flag up #1080 at this point in case it fits in with what you're thinking about already

Reply:
we can pretty easily do a context manager to solve that issue, easier perhaps once we merge this?